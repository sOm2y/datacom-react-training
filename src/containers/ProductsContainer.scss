.trigger {
  font-size: 30px;
  line-height: 64px;
  padding: 0 24px;
  cursor: pointer;
  transition: color 0.3s;
  float: right;
  &:hover {
    color: #1890ff;
  }
}

.ant-badge-count, .ant-badge-dot, .ant-badge .ant-scroll-number-custom-component{
  top:15px;
}