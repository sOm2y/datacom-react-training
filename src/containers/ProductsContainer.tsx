import React from 'react'
import { Layout, Icon, Row, Col } from 'antd'
import { Product } from '../entities/Product'
import { ProductCard } from '../components/ProductCard'
const { Header, Content } = Layout

interface ProductsContainerProps {
    collapsed: boolean,
    products?: Product[],
    toggleCart: () => void
}

export const ProductsContainer: React.FC<ProductsContainerProps> = ({ ...props }) => {
    const { products, toggleCart } = props

    return (
        <>
            <Header style={{ background: '#fff', padding: 0 }}>
                <Icon className="trigger" type="shopping-cart" onClick={toggleCart} />
            </Header>

            <Content
                style={{
                    margin: '24px 16px',
                    padding: 24,
                    background: '#fff',
                    minHeight: 280
                }}
            >
                <Row type="flex">
                    {products &&
                        products.map((product: Product, key: number) => (
                            <Col key={key} span={6}>
                                <ProductCard product={product} />
                            </Col>
                        ))}
                </Row>
            </Content>
        </>
    )
}
