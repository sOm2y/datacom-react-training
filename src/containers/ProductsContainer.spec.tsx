import React from 'react'
import { shallow } from 'enzyme'
import configureStore from 'redux-mock-store'
import ProductsContainer from './ProductsContainer'

describe('Component test for <ProductsContainer />', () => {
	const mockedState = {
		productReducer: { products: [{ name: 'Axe', price: 190.5 }] },
		cartReducer: {
			carts: [{ product: { name: 'Axe', price: 190.5 }, quantity: 1 }]
		}
	}
	const mockStore = configureStore()
	let store, component:any
	beforeEach(() => {
		store = mockStore(mockedState)
		component = shallow(<ProductsContainer store={store} />)
	})
	it('should display product container component', () => {
        expect(component.length).toBe(1)
        expect(component).toMatchSnapshot()
	})
})
