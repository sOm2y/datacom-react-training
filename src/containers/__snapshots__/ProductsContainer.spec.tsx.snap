// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Component test for <ProductsContainer /> should display product container component 1`] = `ShallowWrapper {}`;
